### PR TITLE
Check Uri for Knowledge Base Creation

### LINKED ISSUE #399: Using a non uri as uri for creating a knowledge base leads to a fatal error
Using a non-uri as uri for creating a knowledge base, there is no error/warning. But it leads to a fatal error after clicking on 'view all resources'.

Checking the uri while creating the knowledge base should suffice to fix this.

### PR DESCRIPTION
**Pull request by shinobu**
This fix checks the URI for knowledge base creation for http:// 

This Pull Request does fix  #399

### CODE REVIEW

File: application/controllers/ModelController.php [hunk 1]
```diff
@@ -511,6 +511,10 @@ public function createAction()
                 );
                 $this->view->errorFlag = true;
                 return;
+            } else if (!Erfurt_Uri::check($newModelUri)) {
+                $this->_owApp->appendMessage(
+                    new OntoWiki_Message('The given String is no legal URI (http://...)', OntoWiki_Message::ERROR)
```
Comment (white-gecko) [inline application/controllers/ModelController.php]: The hint "http://…" connotes, that the algorithm already identified the missing protocol part, while there might be other cases for the value not being a valid URI or IRI.

Comment (white-gecko) [inline application/controllers/ModelController.php]: "String" is a very technical term in this case, better use "value"

Comment (white-gecko) [inline application/controllers/ModelController.php]: Since we are aiming for IRI support, it is better to write IRI or URI (resp. other way round)

Comment (white-gecko) [reply application/controllers/ModelController.php]: Also check if the check supports IRIs

**Review by white-gecko — CHANGES_REQUESTED** [review]
The wording of the error message should be improved. Further it would be good to implement the fix in Erfurt and catch the according exception here.
Please also check at which level OntoWiki is already catching such Exceptions.

**Comment by shinobu** [comment]
done in https://github.com/AKSW/Erfurt/pull/129 --- is the error message ok?

**Comment by shinobu** [comment]
now uses error codes to check if the exception gets catched or not

**Comment by white-gecko** [comment]
Do URNs (urn:…) and https:// URI still work?

**Comment by shinobu** [comment]
are urn:'s really valid RDF/Modelnames?

*edit
this change will make them viable choices as well

**Comment by white-gecko** [comment]
What about ftp://… and tel:… I think there are more valid kinds of URIs resp. IRIs and afaik all of them are allowed for models.

https://tools.ietf.org/html/rfc3986
https://tools.ietf.org/html/rfc3987


https://en.wikipedia.org/wiki/Uniform_Resource_Identifier
https://en.wikipedia.org/wiki/Internationalized_Resource_Identifier

**Comment by shinobu** [comment]
We are now using the PHP filter `FILTER_VALIDATE_URL`, We might need to add a test (the ModelController has no tests at all right now) to check if the filter works for all PHP versions though.

**Comment by white-gecko** [comment]
Could you add the test?

File: application/tests/integration/controller/ModelControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            ['news:comp.aksw.www.servers.unix', true],
+            ['telnet://melvyl.ucop.example.edu/', true],
+            ['http://www.ietf.org/rfc/rfc2396.txt', true],
+            ['ldap://[2001:db8::7]/c=GB?objectClass?one', true],
+            ['mailto:AKSW.L@example.com', true],
+            ['telnet://192.0.2.16:80/', true],
+            ['urn:oasis:names:specification:docbook:dtd:xml:4.1.2', true],
+            ['https://www.aksw.org/faq', true],
+            ['ptth://www.aksw.org', true],
+            ['\\äüö', false]
```
Comment (white-gecko) [inline application/tests/integration/controller/ModelControllerTest.php]: Please add some more failing examples

File: application/tests/integration/controller/ModelControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        $this->frontEndLogin();
+    }
+
+    /**
+     * @dataProvider uriProvider
+     */
+    public function testCreateActionFiltersOnlyIncorrectUris($uri, $correctUri)
+    {
+        
+
```
Comment (white-gecko) [inline application/tests/integration/controller/ModelControllerTest.php]: Please also check the coding standard with code sniffer

**Review by white-gecko — CHANGES_REQUESTED** [review]
Finally please rebase and squash the commits.

File: application/controllers/ModelController.php [hunk 4]
```diff
@@ -499,17 +499,33 @@ public function createAction()
             if ($store->isModelAvailable($newModelUri, false)) {
                 // model exists
                 $this->_owApp->appendMessage(
-                    new OntoWiki_Message('Given Knowledge Base already exists.', OntoWiki_Message::ERROR)
+                    new OntoWiki_Message(
+                        'Given Knowledge Base already exists.',
+                        OntoWiki_Message::ERROR
+                    )
```
Comment (white-gecko) [inline application/controllers/ModelController.php]: Why do you touch this twice?

File: application/controllers/ModelController.php [hunk 5]
```diff
@@ -2,7 +2,7 @@
 /**
  * This file is part of the {@link http://ontowiki.net OntoWiki} project.
  *
- * @copyright Copyright (c) 2006-2016, {@link http://aksw.org AKSW}
+ * @copyright Copyright (c) 2006-2017, {@link http://aksw.org AKSW}
  * @license   http://opensource.org/licenses/gpl-license.php GNU General Public License (GPL)
```
Comment (white-gecko) [inline application/controllers/ModelController.php:6]: This can go to the first commit.

**Review by white-gecko — CHANGES_REQUESTED** [review]

**Comment by shinobu** [comment]
Done @white-gecko 
This should be mergeable now

**Review by white-gecko — APPROVED** [review]

**Pull request merged**